General IP intel tweaks

optimumtact opened the pull request.

optimumtact committed:
Move the ipintel verbs to a subtab under admin

optimumtact committed:
Fixes

optimumtact committed:
Wat

optimumtact committed:
Fix string

optimumtact committed:
Add test merge bail out

optimumtact commented:
This should be safe to test merge without ill effect.

optimumtact committed:
Log response body

optimumtact committed:
Tf

ZephyrTFA committed:
Update ipintel.dm

ZephyrTFA committed:
move check intel down

ZephyrTFA committed:
fix admin check

ZephyrTFA committed:
Merge branch 'master' into clussy2

ZephyrTFA committed:
g

MrStonedOne commented on code/controllers/subsystem/ipintel.dm:
```diff
@@ -178,6 +177,8 @@ SUBSYSTEM_DEF(ipintel)
 	return TRUE
 
 /datum/controller/subsystem/ipintel/proc/is_exempt(client/player)
+	if(player.holder)
+		return TRUE
```
```suggestion
	if(player.holder || GLOB.deadmins[player.ckey])
		return TRUE
```

ZephyrTFA committed:
Update code/controllers/subsystem/ipintel.dm

Co-authored-by: Kyle Spier-Swenson <kyleshome@gmail.com>

MrsTonedOne committed:
Fix a bunch of things with ipintel.

Made the db code better.
Made the config validation code better.
Made everything check if the config is valid before doing things.
Removed cached copies of the config from the datum.
Tweaked anything else that seemed off.

MrsTonedOne committed:
improve ipintel config validation, remove now unused config

These checks go here.

optimumtact committed:
Merge pull request #12 from MrStonedOne/patch-673

Fix a bunch of things with ipintel.

MrStonedOne commented on code/controllers/configuration/entries/general.dm:
```diff
@@ -453,7 +453,7 @@
 
 /datum/config_entry/string/ipintel_email/ValidateAndSet(str_val)
-	return str_val != "ch@nge.me" && ..()
+	return str_val != "ch@nge.me" && (!length(str_var) || findtext(str_val, "@")) && ..()
```
```suggestion
	return str_val != "ch@nge.me" && (!length(str_val) || findtext(str_val, "@")) && ..()
```

MrStonedOne commented on code/controllers/subsystem/ipintel.dm:
```diff
@@ -30,47 +18,39 @@ SUBSYSTEM_DEF(ipintel)
-		fail_messages += list("invalid probability threshold")
-
+	
 	contact_email = CONFIG_GET(string/ipintel_email)
```
```suggestion
	var/contact_email = CONFIG_GET(string/ipintel_email)
```

MrStonedOne commented on code/controllers/subsystem/ipintel.dm:
```diff
@@ -109,17 +89,18 @@ SUBSYSTEM_DEF(ipintel)
 
-	var/query_base = "https://[src.query_base]/check.php?ip="
+	var/query_base = "https://[CONFIG_GET(string/ipintel_base)]/check.php?ip="
 	var/query = "[query_base][address]&contact=[contact_email]&flags=b&format=json"
```
```suggestion
	var/query = "[query_base][address]&contact=[CONFIG_GET(string/ipintel_email)]&flags=b&format=json"
```

MrStonedOne commented on code/controllers/subsystem/ipintel.dm:
```diff
@@ -150,13 +134,17 @@ SUBSYSTEM_DEF(ipintel)
-		date_restrictor = " AND date > DATE_SUB(NOW(), INTERVAL [ipintel_cache_length] DAY)"
+	if (!SSdbcore.Connect())
+		return
+	ipintel_cache_length = CONFIG_GET(number/ipintel_cache_length)
```
```suggestion
	var/ipintel_cache_length = CONFIG_GET(number/ipintel_cache_length)
```

MrStonedOne commented on code/controllers/subsystem/ipintel.dm:
```diff
@@ -199,9 +190,13 @@ SUBSYSTEM_DEF(ipintel)
+	
 
 ADMIN_VERB(ipintel_allow, R_BAN, "Whitelist Player VPN", "Allow a player to connect even if they are using a VPN.", ADMIN_CATEGORY_IPINTEL, ckey as text)
+	if (!is_enabled())
```
```suggestion
	if (!SSipintel.is_enabled())
```

MrStonedOne commented on code/controllers/subsystem/ipintel.dm:
```diff
@@ -224,6 +219,8 @@ ADMIN_VERB(ipintel_allow, R_BAN, "Whitelist Player VPN", "Allow a player to conn
 	message_admins("IPINTEL: [key_name_admin(user)] has whitelisted '[ckey]'")
 
 ADMIN_VERB(ipintel_revoke, R_BAN, "Revoke Player VPN Whitelist", "Revoke a player's VPN whitelist.", ADMIN_CATEGORY_IPINTEL, ckey as text)
+	if (!is_enabled())
```
```suggestion
	if (!SSipintel.is_enabled())
```

MrStonedOne commented on code/controllers/subsystem/ipintel.dm:
```diff
@@ -238,6 +235,8 @@ ADMIN_VERB(ipintel_revoke, R_BAN, "Revoke Player VPN Whitelist", "Revoke a playe
 	message_admins("IPINTEL: [key_name_admin(user)] has revoked the VPN whitelist for '[ckey]'")
 
 /client/proc/check_ip_intel()
+	if (!is_enabled())
```
```suggestion
	if (!SSipintel.is_enabled())
```

MrsTonedOne reviewed (changes requested):
Missed some things, made some typos.

optimumtact committed:
Update code/controllers/configuration/entries/general.dm

Co-authored-by: Kyle Spier-Swenson <kyleshome@gmail.com>

optimumtact committed:
Update code/controllers/subsystem/ipintel.dm

Co-authored-by: Kyle Spier-Swenson <kyleshome@gmail.com>

optimumtact committed:
Update code/controllers/subsystem/ipintel.dm

Co-authored-by: Kyle Spier-Swenson <kyleshome@gmail.com>

optimumtact committed:
Update code/controllers/subsystem/ipintel.dm

Co-authored-by: Kyle Spier-Swenson <kyleshome@gmail.com>

optimumtact committed:
Update code/controllers/subsystem/ipintel.dm

Co-authored-by: Kyle Spier-Swenson <kyleshome@gmail.com>

optimumtact committed:
Update code/controllers/subsystem/ipintel.dm

Co-authored-by: Kyle Spier-Swenson <kyleshome@gmail.com>

optimumtact committed:
Update code/controllers/subsystem/ipintel.dm

Co-authored-by: Kyle Spier-Swenson <kyleshome@gmail.com>

MrStonedOne commented:
Set living to 5 hours, it complained about an error on the one person who had less than that but didn't check anybody else. likely we got the outgoing ip blacklisted, easy fix since i have 5 of them. anyways the rest of the code works so the todo return can be removed.

MrStonedOne commented on code/controllers/subsystem/ipintel.dm:
```diff
@@ -273,9 +272,11 @@ ADMIN_VERB(ipintel_revoke, R_BAN, "Revoke Player VPN Whitelist", "Revoke a playe
 			else
 				message_admins("IPINTEL: [key_name_admin(src)] was unable to be checked due to an error.")
 
+	// TODO remove after test merge
+	return
 	if(!connection_rejected)
```
```suggestion
				message_admins("IPINTEL: [key_name_admin(src)] was unable to be checked due to an error.")

	if(!connection_rejected)
```

ZephyrTFA committed:
Update code/controllers/subsystem/ipintel.dm

Co-authored-by: Kyle Spier-Swenson <kyleshome@gmail.com>

MrStonedOne commented on code/controllers/subsystem/ipintel.dm:
```diff
@@ -1,26 +1,14 @@
 	var/list/datum/ip_intel/cached_queries = list()
 	/// The store for rate limiting
-	var/list/rate_limits
+	var/list/rate_limit_minute = list()
```
```suggestion
	var/list/rate_limit_minute
```

MrStonedOne replied:
this one was my fuck up

MrStonedOne commented:
Ok no, its not the ratelisting. I was able to simulate a query in chrome on sybil's vm and it ran just fine, but so far its not successfully making queries in game.

We'll need to add some debug logging to the actual requests.

MrStonedOne commented on code/controllers/subsystem/ipintel.dm:
```diff
@@ -109,17 +89,18 @@ SUBSYSTEM_DEF(ipintel)
 	if(!initialized)
 		return IPINTEL_UNKNOWN_INTERNAL_ERROR
 
```
```suggestion

```

optimumtact committed:
Update code/controllers/subsystem/ipintel.dm

Co-authored-by: Kyle Spier-Swenson <kyleshome@gmail.com>

optimumtact commented:
> Ok no, its not the ratelisting. I was able to simulate a query in chrome on sybil's vm and it ran just fine, but so far its not successfully making queries in game.
> 
> We'll need to add some debug logging to the actual requests.

did my debug log not work.

MrStonedOne commented:
it wasn't initializing so it never reached the debug line, which is how i figured out what it was

optimumtact committed:
Update code/controllers/subsystem/ipintel.dm

Co-authored-by: Kyle Spier-Swenson <kyleshome@gmail.com>

MrStonedOne commented:
now the debug logs are printing.

Its a blank request body. Likely gonna need to see the http status code to understand. Maybe even the response headers.

MrStonedOne commented on code/controllers/subsystem/ipintel.dm:
```diff
@@ -92,34 +72,32 @@ SUBSYSTEM_DEF(ipintel)
 	request.prepare(RUSTG_HTTP_METHOD_GET, query)
 	request.execute_blocking()
 	var/datum/http_response/response = request.into_response()
 	var/list/data = response.body
```
```suggestion
	var/list/data = json_decode(response.body)
```

optimumtact committed:
Update code/controllers/subsystem/ipintel.dm

Co-authored-by: Kyle Spier-Swenson <kyleshome@gmail.com>

jlsnow301 commented:
does this close #83008

ZephyrTFA committed:
Merge branch 'master' into clussy2

MrStonedOne commented:
> does this close #83008

Likely, i added some db status checks to ipintel but would need somebody to test first
